### PR TITLE
[NET-468] Feature object counter in category sidebar

### PR DESCRIPTION
**Pull request by doblanken**
Updates the object counter in the category sidebar when objects are added or deleted.
Resolves the issue stated in [NET-468](https://nethinks.atlassian.net/browse/NET-468)

### CODE REVIEW

File: app/src/app/layout/services/sidebar.service.ts [hunk 1]
```diff
@@ -51,4 +55,47 @@ export class SidebarService {
...
+  }
+
+  /**
+   * updates the objectCounter of the affected sidebar-type component
+   *
+   * @param typeID the type id with to the sidebar-typ component will be filtered
+   */
+  public async updateTypeCounter(typeID) {
+    const sidebarType = this.sideBarType.filter(type => type.type.public_id === typeID).pop();
+    sidebarType.objectCounter = await this.getObjectCount(sidebarType.type.public_id);
```
Comment (markheumueller) [inline app/src/app/layout/services/sidebar.service.ts:79]: You store a promise in a number field here.

**Review by markheumueller — CHANGES_REQUESTED** [review]
Hey Dorus, please look again with TSLint over it. I get several errors displayed there.

**Comment by doblanken** [comment]
> Hey Dorus, please look again with TSLint over it. I get several errors displayed there.

Could you please define which errors are displayed? I don't see any here.

**Review by markheumueller — APPROVED** [review]
LGTM

**Pull request merged**